python/sparkdl/transformers/tf_image.py

```diff
@@ -47,42 +53,39 @@ class TFImageTransformer(Transformer, HasInputCol, HasOutputCol, HasOutputMode):
               since a new session is created inside this transformer.
     """
 
-    USER_GRAPH_NAMESPACE = 'given'
-    NEW_OUTPUT_PREFIX = 'sdl_flattened'
-
     graph = Param(Params._dummy(), "graph", "A TensorFlow computation graph",
                   typeConverter=SparkDLTypeConverters.toTFGraph)
     inputTensor = Param(Params._dummy(), "inputTensor",
                         "A TensorFlow tensor object or name representing the input image",
-                        typeConverter=SparkDLTypeConverters.toStringOrTFTensor)
+                        typeConverter=SparkDLTypeConverters.toTFTensorName)
     outputTensor = Param(Params._dummy(), "outputTensor",
                          "A TensorFlow tensor object or name representing the output",
-                         typeConverter=SparkDLTypeConverters.toStringOrTFTensor)
+                         typeConverter=SparkDLTypeConverters.toTFTensorName)
 
     @keyword_only
     def __init__(self, inputCol=None, outputCol=None, graph=None,
-                 inputTensor=utils.IMAGE_INPUT_PLACEHOLDER_NAME, outputTensor=None,
+                 inputTensor=IMAGE_INPUT_TENSOR_NAME, outputTensor=None,
                  outputMode="vector"):
         """
         __init__(self, inputCol=None, outputCol=None, graph=None,
-                 inputTensor=utils.IMAGE_INPUT_PLACEHOLDER_NAME, outputTensor=None,
+                 inputTensor=IMAGE_INPUT_TENSOR_NAME, outputTensor=None,
                  outputMode="vector")
         """
         super(TFImageTransformer, self).__init__()
-        self._setDefault(inputTensor=utils.IMAGE_INPUT_PLACEHOLDER_NAME)
```

Comment:
best off putting them into `setParams` so that it is clear they are set there.

Reply:
setting `inputTensor=utils.IMAGE_INPUT_PLACEHOLDER_NAME` will cause TensorFlow to throw an error since `utils.IMAGE_INPUT_PLACEHOLDER_NAME` cannot be interpreted as a Tensor name.

Reply:
Did you run into a problem with tf_image.py in this PR or tests? Not sure why this change was made here.

Reply:
What happens with `inputTensor` if someone never uses setParams? Seems breakable.

Reply:
Previously, if anyone chose not to set `inputTensor` (either in the constructor or using the setter method), the default `utils.IMAGE_INPUT_PLACEHOLDER_NAME` will be used. (In fact, the default is always set to this value).

Later on, we will call `self.getGraph().get_tensor_by_name(tensor_name)`, in which case if the default value is used, will cause TensorFlow to throw an exception. 

It happens that all of our code set `inputTensor`. But if our users choose to use interact directly with `TFImageTransformer` and use the default, they will encounter that exception.

Reply:
Sorry I missed that `__init__` calls `setParams` so defaults should always be set and it's okay.